source/common/http/conn_manager_utility.cc

```diff
@@ -107,8 +110,17 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
         Utility::appendXff(request_headers, *connection.remoteAddress());
       }
     }
-    request_headers.insertForwardedProto().value().setReference(
-        connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
+    if (trusted_forwarded_proto) {
+      // Set x-forwarded-proto unless it already exists and was forwarded on by a trusted proxy.
```

Comment:
nit: can you rephrase this comment to read more in line with the logic that follows (i.e. or vs. and, etc.). Normally I wouldn't be picky like this but this code is so hard to understand as it is that any comments hopefully should follow pretty closely. (I'm having a bit of a hard time groking what is happening here.)

Reply:
Sure.  Also note that I could just overwrite if num_trusted_hops is 0 and allow the logic at :139 to set proto if absent if you prefer that.  I felt it was more clear to have it in one place than fall through, but that may just be me :-)